libs/payments/management/src/lib/subscriptionManagement.service.ts

```diff
@@ -67,10 +92,158 @@ export class SubscriptionManagementService {
           subs,
           uid
         );
+
+      accountCreditBalance = {
+        balance: Math.abs(customer.balance),
+        currency: customer.currency ?? subs[0]?.currency,
+      };
+
+      if (subs.length === 0) {
+        return { accountCreditBalance, defaultPaymentMethod, subscriptions };
+      }
+
+      const priceIds = subs.flatMap((sub) =>
+        sub.items.data.map((item) => item.price.id)
+      );
+      const productMap =
+        await this.productConfigurationManager.getProductNameByPriceIds(
+          priceIds
+        );
+
+      if (!productMap) {
+        throw new SubscriptionManagementCouldNotRetrieveProductNamesFromCMSError(
+          priceIds
+        );
+      }
+
+      const subsContent = await Promise.all(
+        subs.map(async (sub) => {
+          try {
+            const item = sub.items.data[0];
+            const price = item.price;
+            const priceId = price.id;
+            const productName = productMap.productNameForPriceId(priceId);
+            if (!productName) {
+              Sentry.captureMessage('No product name for price id', {
+                extra: { priceId },
+              });
+              throw new SubscriptionManagementNoProductNameForPriceIdError(
+                priceId
+              );
+            }
+            return this.getSubscriptionContent(
+              sub,
+              customer,
+              price,
+              productName ?? 'Mozilla Subscription'
```

Comment:
Since you're throwing an error above if productName isn't present this ?? isn't necessary